regression_tests/utils.go

```diff
@@ -0,0 +1,305 @@
+}
+
+// getSharedConfig retrieves the shared configuration from the SPV Wallet.
+func getSharedConfig(xpub string) (*models.SharedConfig, error) {
+	req, err := http.NewRequest(http.MethodGet, addPrefixIfNeeded(domainLocalHost)+domainSuffixSharedConfig, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	req.Header.Set(models.AuthHeader, xpub)
+	client := &http.Client{}
+	resp, err := client.Do(req)
+	if err != nil {
+		return nil, err
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode != http.StatusOK {
+		return nil, fmt.Errorf("failed to get shared config: %s", resp.Status)
+	}
+
+	body, err := io.ReadAll(resp.Body)
+	if err != nil {
+		return nil, err
+	}
+
+	var configResponse models.SharedConfig
+	if err := json.Unmarshal(body, &configResponse); err != nil {
+		return nil, err
+	}
+
+	if len(configResponse.PaymailDomains) != 1 {
+		return nil, fmt.Errorf("expected 1 paymail domain, got %d", len(configResponse.PaymailDomains))
+	}
+	return &configResponse, nil
+}
+
+// promptUserAndCheck prompts the user with a question and validates the response.
+func promptUserAndCheck(question string) int {
+	reader := bufio.NewReader(os.Stdin)
+	var response string
+	var checkResult int
+
+	for {
+		fmt.Println(question)
+		response, _ = reader.ReadString('\n')
+		response = strings.TrimSpace(response)
+
+		checkResult = checkResponse(response)
+		if checkResult != -1 {
+			break
+		}
+		fmt.Println("Invalid response. Please answer y/yes or n/no.")
+	}
+
+	return checkResult
+}
+
+// checkResponse checks the response and returns an integer indicating the result.
+func checkResponse(response string) int {
+	response = strings.ToLower(strings.TrimSpace(response))
+	switch response {
+	case "yes", "y":
+		return 1
+	case "no", "n":
+		return 0
+	default:
+		return -1
+	}
+}
+
+// preparePaymail constructs a paymail address from the alias and domain.
+func preparePaymail(paymailAlias string, domain string) string {
+	return paymailAlias + atSign + domain
+}
+
+// createUser creates a new user with the specified paymail.
+func createUser(paymail string, config *regressionTestConfig) (*regressionTestUser, error) {
+	keys, err := xpriv.Generate()
+	if err != nil {
+		return nil, err
+	}
+
+	user := &regressionTestUser{
+		XPriv:   keys.XPriv(),
+		XPub:    keys.XPub().String(),
+		Paymail: preparePaymail(paymail, config.ClientOneURL),
+	}
+
+	adminClient := walletclient.NewWithAdminKey(addPrefixIfNeeded(domainLocalHost), adminXPriv)
+	ctx := context.Background()
+
+	if err := adminClient.AdminNewXpub(ctx, user.XPub, map[string]any{"some_metadata": "remove"}); err != nil {
+		fmt.Println("adminNewXpub failed with status code:", err)
+		return nil, err
+	}
+
+	createPaymailRes, err := adminClient.AdminCreatePaymail(ctx, user.XPub, user.Paymail, "Test test", "")
+	if err != nil {
+		if err.Error() == "paymail address already exists" {
+			return user, fmt.Errorf("paymail address already exists")
+		}
+		return nil, err
+	}
+
+	fmt.Println(keys.XPriv())
```

Comment:
Does this needed ?

Reply:
Can be useful to easily reuse the "leader" account.